### PR TITLE
[bloomsky] BloomSky Binding initial contribution

### PR DESCRIPTION
**Pull request by dschoepel**
Signed-off-by: Dave J Schoepel <dave@theschoepels.com>

[bloomsky] Initial contribution

Implementing an OpenHab binding for the BloomSky personal weather station.  This is a "read-only" binding that uses the [BloomSky API](http://weatherlution.com/bloomsky-api/?doing_wp_cron=1615241711.4678061008453369140625) to retrieve the sensor data from the **SKY1, SKY2** and **STORM** personal weather stations.  If you are not familiar with these weather stations, you can find out more at the [BloomSky Home Page](https://www.bloomsky.com/).

This pull request includes the initial working binding that has been developed against the 3.2.0-SNAPSHOT.  The coding guidelines have been followed, all issues from the static code analysis have been addressed.    

Link to JAR file: https://nextcloud.theschoepels.com/s/RHq2fGeE49dQ3Zg

This binding has been tested against the developer's BloomSky account that has two Sky's and one Storm weather station associated with it.  It is currently running in a OH 3.2.0 system in both Windows and RPI.

### CODE REVIEW

**Comment by dschoepel** [comment]
New to this process and the Jenkins build seems to have failed.  Checking the console log there seems to be a problem with my binding not passing a spotless check along with several other bindings that are not part of my request.  I ran the spotless check prior to commit and this pull request and had no errors.  I re-ran the check and still do not see any errors.  Am I missing something?  Any help /advice is appreciated....

File: bundles/pom.xml [hunk 1]
```diff
@@ -60,7 +60,8 @@
     <module>org.openhab.binding.avmfritz</module>
     <module>org.openhab.binding.benqprojector</module>
     <module>org.openhab.binding.bigassfan</module>
-    <module>org.openhab.binding.bluetooth</module>
+    <module>org.openhab.binding.bloomsky</module>
+	<module>org.openhab.binding.bluetooth</module>
```
Comment (dalgwen) [inline bundles/pom.xml:64]: It seems you used on the bluetooth line a "\t" instead of the usual four spaces, thus the spotless check failing.

Comment (dschoepel) [reply bundles/pom.xml]: Thanks for the help  @dalgwen  - so simple...

**Comment by dschoepel** [comment]
@Skinah , The code-analysis reports are clean/empty (no warnings), and both checks have passed. Let me know if there is anything else I need to do...

File: bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+		<label>Wind Gust</label>
+		<description>Wind gust</description>
+		<category>Wind</category>
+		<state readOnly="true" pattern="%.1f %unit%"/>
+	</channel-type>
+
+	<channel-type id="windChill">
+		<item-type>Number:Temperature</item-type>
+		<label>Wind Chill</label>
+		<description>Wind chill in °F or °C</description>
```
Comment (Skinah) [inline bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml:534]: remove any referrence to C or F as it is automatic due to your using UOM.

File: bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+		<item-type>Number:Length</item-type>
+		<label>Precipitation 24h</label>
+		<description>Precipitation last 24 hours</description>
+		<category>Rain</category>
+		<state readOnly="true" pattern="%.2f %unit%"/>
+	</channel-type>
+
+	<channel-type id="rainDaily">
+		<item-type>Number:Length</item-type>
+		<label>Precipitation Total Past 24h</label>
```
Comment (Skinah) [inline bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml:509]: We prefer the label to be 2-3 words max. Suggest:
Todays Rain
Rain Today
Rain Daily

I tend to prefer keeping it the same as the channel ID as it makes it easier for the textual people. The label is used to auto create the name for the items when you create equipment from a thing.

**Comment by dschoepel** [comment]
@Skinah , I could use some advice on how to correct the DCO check failure.  I am not clear on what steps I need to take to fix the following error?  Thanks in advance....

Commit sha: b1c5108, Author: lolodomo, Committer: Dave J Schoepel; Expected "lolodomo lg.hc@free.fr", but got "Laurent Garnier lg.hc@free.fr".

File: bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml (hunk 3 above)
Comment (dschoepel) [reply bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml]: Went through entire list of labels and made changes as suggested.

File: bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (dschoepel) [reply bundles/org.openhab.binding.bloomsky/src/main/resources/OH-INF/thing/thing-types.xml:534]: Removed temp indicator as suggested.

**Comment by dschoepel** [comment]
I need to close this pull request and open a new one as this branch has been corrupted by my efforts to get it cleaned up after doing a rebase out of sequence (still a GIT rookie).  I will open a new one and reference this so we don't lose the conversation.

**Comment by dschoepel** [comment]
Bloomsky seems to have gone out of business and abandoned its customers.  The API is no longer accessible.  Removed this pull request and moved on to a different weather station.  Tempest by Weatherflow.

**Pull request closed without merging**